extensions/InputBox/InputBox.hooks.php

```diff
@@ -52,24 +52,38 @@ public static function onMediaWikiPerformAction(
 		$request, 
 		$wiki )
 	{
+		global $wgScript, $wgUser;
+
 		if( $wiki->getAction( $request ) !== 'edit' ){
 			# not our problem
 			return true;
 		}
+
+		$params = $request->getValues();
+
+		if ( $request->getText( 'preload', '' ) !== '' ){
+			# Make sure any call with preload uses VE when it matches the user's prefs
+			if ( EditorPreference::isVisualEditorPrimary() && $wgUser->isLoggedIn() ) {
```

Comment:
You can avoid the use of the global `$wgUser` as the current user is passed to this method in the `$user` variable.